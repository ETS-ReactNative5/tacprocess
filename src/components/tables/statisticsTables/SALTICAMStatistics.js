import React from "react";
import propTypes from "prop-types";

function countSCAMDetector(proposals) {
	let expo = {
		dscan: 0,
		xfer: 0,
		normal: 0,
		shuffle: 0,
		slot: 0,
	};
	(proposals|| []).forEach(p => {
		p.instruments.rss.forEach( m => {
			if (m.detectorMode === "NORMAL") {expo.normal += 1}
			if (m.detectorMode === "SLOT") {expo.slot += 1}
			if (m.detectorMode === "DRIFTSCAN") {expo.dscan += 1}
			if (m.detectorMode === "FRAME XFER") {expo.xfer += 1}
		})
	});
	return expo
}

const SALTICAMStatistics = (proposals) => {
	const detectorMode = countSCAMDetector(proposals);
	return(
		<div>
			{/* Salticam table */}
			<h2>Salticam</h2>
			<table className="table center-table">
				<thead>
				<tr>
					<th><h2>Detector mode</h2></th>
					<th><h2>Number of Proposals</h2></th>
				</tr>
				</thead>
				<tbody>
				<tr>
					<td><h2>DRIFT SCAN</h2></td>
					<td><h2>{ detectorMode.dscan }</h2></td>
				</tr>
				<tr>
					<td><h2>FRAME XFER</h2></td>
					<td><h2>{ detectorMode.xfer }</h2></td>
				</tr>
				<tr>
					<td><h2>NORMAL</h2></td>
					<td><h2>{ detectorMode.normal }</h2></td>
				</tr>
				<tr>
					<td><h2>SLOT</h2></td>
					<td><h2>{ detectorMode.slot }</h2></td>
				</tr>
				</tbody>
			</table>
		</div>
	)
};

SALTICAMStatistics.propTypes = {
	proposals: propTypes.array.isRequired,
};

export default SALTICAMStatistics;