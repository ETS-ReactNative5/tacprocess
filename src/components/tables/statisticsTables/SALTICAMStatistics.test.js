import React from "react";
import renderer from "react-test-renderer";
import { shallow, mount, render } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import SALTICAMStatistics from "./SALTICAMStatistics";

const proposals = [
  {
    transparency: 'Clear',
    maxSeeing: 2,
    instruments: {
      hrs: [ { exposureMode: "HIGH RESOLUTION" } ],
      rss: [ { dictatorMode: 'NORMAL', mode: 'Spectroscopy' } ],
      scam: [ { dictatorMode: "NORMAL" } ],
      bvit: [{type: "BVIT"}]
    },
    timeRequests: [
      {
        semester: '2018-1',
        distribution: [ { partnerCode: 'RSA', time: 150 }, { partnerCode: 'UW', time: 200 }, { partnerCode: 'IUCAA', time: 250 } ]
      },
      {
        semester: '2018-1',
        distribution: [ { partnerCode: 'RSA', time: 400 }, { partnerCode: 'UW', time: 50 } ]
      }
    ]
  },
  {
    transparency: 'Thin cloud',
    maxSeeing: 2,
    instruments: {
      hrs: [ { exposureMode: "HIGH RESOLUTION" } ],
      rss: [ { dictatorMode: 'NORMAL', mode: 'Spectroscopy' } ],
      scam: [ { dictatorMode: "NORMAL" } ],
      bvit: [{type: "BVIT"}]
    },
    timeRequests: [
      {
        semester: '2017-1',
        distribution: [ { partnerCode: 'RSA', time: 150 }, { partnerCode: 'UW', time: 200 }, { partnerCode: 'IUCAA', time: 250 } ]
      },
      {
        semester: '2018-1',
        distribution: [ { partnerCode: 'RSA', time: 400 }, { partnerCode: 'UW', time: 50 } ]
      }
    ]
  }
];

// Checking if the SALTICAMStatistics Component renders correctly for different input
describe("SALTICAMStatistics Component", () => {

  //Enzyme method testing if it renders correctly with empty values of the proposals, semester, and partner
  it("Render the SALTICAMStatistics Component having unpopulated props with no errors", () => {
    const rendered = mount(<SALTICAMStatistics proposals={[]}  />);
    expect(shallowToJson(rendered)).toMatchSnapshot();
  });

  it("Render the SALTICAMStatistics Component having populated props with no errors", () => {
    const rendered = mount(<SALTICAMStatistics proposals={proposals}  />);
    expect(shallowToJson(rendered)).toMatchSnapshot();
  });
});
