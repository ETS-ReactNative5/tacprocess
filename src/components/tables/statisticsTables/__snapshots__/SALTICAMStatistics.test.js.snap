// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SALTICAMStatistics Component Render the SALTICAMStatistics Component having populated props with no errors 1`] = `
<SALTICAMStatistics
  proposals={
    Array [
      Object {
        "instruments": Object {
          "bvit": Array [
            Object {
              "type": "BVIT",
            },
          ],
          "hrs": Array [
            Object {
              "exposureMode": "HIGH RESOLUTION",
            },
          ],
          "rss": Array [
            Object {
              "dictatorMode": "NORMAL",
              "mode": "Spectroscopy",
            },
          ],
          "scam": Array [
            Object {
              "dictatorMode": "NORMAL",
            },
          ],
        },
        "maxSeeing": 2,
        "timeRequests": Array [
          Object {
            "distribution": Array [
              Object {
                "partnerCode": "RSA",
                "time": 150,
              },
              Object {
                "partnerCode": "UW",
                "time": 200,
              },
              Object {
                "partnerCode": "IUCAA",
                "time": 250,
              },
            ],
            "semester": "2018-1",
          },
          Object {
            "distribution": Array [
              Object {
                "partnerCode": "RSA",
                "time": 400,
              },
              Object {
                "partnerCode": "UW",
                "time": 50,
              },
            ],
            "semester": "2018-1",
          },
        ],
        "transparency": "Clear",
      },
      Object {
        "instruments": Object {
          "bvit": Array [
            Object {
              "type": "BVIT",
            },
          ],
          "hrs": Array [
            Object {
              "exposureMode": "HIGH RESOLUTION",
            },
          ],
          "rss": Array [
            Object {
              "dictatorMode": "NORMAL",
              "mode": "Spectroscopy",
            },
          ],
          "scam": Array [
            Object {
              "dictatorMode": "NORMAL",
            },
          ],
        },
        "maxSeeing": 2,
        "timeRequests": Array [
          Object {
            "distribution": Array [
              Object {
                "partnerCode": "RSA",
                "time": 150,
              },
              Object {
                "partnerCode": "UW",
                "time": 200,
              },
              Object {
                "partnerCode": "IUCAA",
                "time": 250,
              },
            ],
            "semester": "2017-1",
          },
          Object {
            "distribution": Array [
              Object {
                "partnerCode": "RSA",
                "time": 400,
              },
              Object {
                "partnerCode": "UW",
                "time": 50,
              },
            ],
            "semester": "2018-1",
          },
        ],
        "transparency": "Thin cloud",
      },
    ]
  }
>
  <div
    className="stat-item"
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    <table
      className="stat-table"
    >
      <thead>
        <tr>
          <th>
            Detector mode
          </th>
          <th>
            Number of Proposals
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            DRIFT SCAN
          </td>
          <td>
            0
          </td>
        </tr>
        <tr>
          <td>
            FRAME XFER
          </td>
          <td>
            0
          </td>
        </tr>
        <tr>
          <td>
            NORMAL
          </td>
          <td>
            0
          </td>
        </tr>
        <tr>
          <td>
            SLOT
          </td>
          <td>
            0
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</SALTICAMStatistics>
`;

exports[`SALTICAMStatistics Component Render the SALTICAMStatistics Component having unpopulated props with no errors 1`] = `
<SALTICAMStatistics
  proposals={Array []}
>
  <div
    className="stat-item"
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    <table
      className="stat-table"
    >
      <thead>
        <tr>
          <th>
            Detector mode
          </th>
          <th>
            Number of Proposals
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            DRIFT SCAN
          </td>
          <td>
            0
          </td>
        </tr>
        <tr>
          <td>
            FRAME XFER
          </td>
          <td>
            0
          </td>
        </tr>
        <tr>
          <td>
            NORMAL
          </td>
          <td>
            0
          </td>
        </tr>
        <tr>
          <td>
            SLOT
          </td>
          <td>
            0
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</SALTICAMStatistics>
`;
